File: src/Poker/hand/ThreeOfAKind.java
package Poker.hand;

import Poker.game.DeckConstants.Value;

public class ThreeOfAKind extends Hand
{
   private Value threeKindValue;
   private Value kickers[];
   
   public ThreeOfAKind(Value values[])
   {
      super(Rank.THREE_KIND);
      threeKindValue = values[0];
      kickers = new Value[2];
      for (int i = 0; i < 2; i++)
      {
         kickers[i] = values[i + 1];
      }
   }
   
   public Value getThreeKindValue()
   {
      return threeKindValue;
   }
   
   public Value getKicker(int i)
   {
      return kickers[i];
   }
   
   public String toString()
   {
      return "3 of a kind - " + threeKindValue.toString() + "s";
   }
   
   public int compare(Hand handToCompare)
   {
      int iCompare = compareRank(handToCompare);
      if (iCompare != 0)
      {
         return iCompare;
      }
      
      return compareThreeOfAKind((ThreeOfAKind) handToCompare);
   }
   
   private int compareThreeOfAKind(ThreeOfAKind tk)
   {
      if (threeKindValue != tk.getThreeKindValue())
      {
         return threeKindValue.getIndex() - tk.getThreeKindValue().getIndex();
      }
      
      for (int i = 0; i < 2; i++)
      {
         if (kickers[i] != tk.getKicker(i))
         {
            return kickers[i].getIndex() - tk.getKicker(i).getIndex();
         }
      }
      
      return 0;
   }
}

File: server/hand/threeOfAKind.js
let Hand = require('./hand');

let ThreeOfAKind = (threeKindValue, kickers) =>
{
	const THREE_KIND = 3;
	
	let rank = Hand(THREE_KIND);
	
	let self =
	{
		compare: (hand) =>
		{
			let rankCompare = rank.compare(hand);
			if (rankCompare)
			{
				return rankCompare;
			}
			
			return self.compareSameRank(hand);
		},
		compareSameRank: (threeOfAKind) =>
		{
			if (threeKindValue !== threeOfAKind.getThreeKindValue())
			{
				return threeKindValue - threeOfAKind.getThreeKindValue();
			}
			
			for (let i = 0; i < 2; i++)
			{
				if (kickers[i] !== threeOfAKind.getKicker(i))
				{
					return kickers[i] - threeOfAKind.getKicker(i);
				}
			}
			
			return 0;
		},
		getThreeKindValue: () =>
		{
			
		},
		getKicker: (i) => kickers[i],
		toString: () =>
		{
			
		}
	}
	
	return self;
}

module.exports = ThreeOfAKind;
